pkg/operator/controller/gardenlet/reconciler.go

```diff
@@ -130,26 +133,43 @@ func (r *Reconciler) reconcile(
 	gardenlet *seedmanagementv1alpha1.Gardenlet,
 	actuator gardenletdeployer.Interface,
 ) (
+	result reconcile.Result,
 	err error,
 ) {
 	status := gardenlet.Status.DeepCopy()
 	status.ObservedGeneration = gardenlet.Generation
 
 	log.V(1).Info("Reconciling")
+	if !r.seedDoesNotExist(ctx, gardenlet) {
+		err = r.cleanupKubeconfigSecret(ctx, log, gardenlet)
+		if err != nil {
+			status.Conditions = updateCondition(r.Clock, status.Conditions, gardencorev1beta1.ConditionFalse, gardencorev1beta1.EventReconcileError, err.Error())
+		} else {
+			log.Info("Reconciliation finished")
+			status.Conditions = updateCondition(r.Clock, status.Conditions, gardencorev1beta1.ConditionTrue, gardencorev1beta1.EventReconciled, fmt.Sprintf("Gardenlet deployed and Seed %q registered", gardenlet.Name))
+		}
```

Comment:
Not sure I understand the log statement here. In case the kubeconfig cleanup fails there will be no `Reconciliation finished` message. However, the reconciliation will probably succeed as the result will be empty and the result of `r.udpateStatus(...)` will likely succeed as well. Hence, I am a bit confused.
Could you please explain?

Reply:
Maybe it's a bit confusing because it wasn't meant to signify the end of a single reconciliation, but rather the end of the entire handling, i.e. Gardenlet deployed, seed registered, kubeconfig cleaned up.

I moved the original message w/o bothering too much but will amend the message for a better understanding.